File: qb-engineer-server/qb-engineer.core/Models/CreateSalesOrderRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateSalesOrderRequestModel(
    int CustomerId,
    int? QuoteId,
    int? ShippingAddressId,
    int? BillingAddressId,
    string? CreditTerms,
    DateTime? RequestedDeliveryDate,
    string? CustomerPO,
    string? Notes,
    decimal TaxRate,
    List<CreateSalesOrderLineModel> Lines);

public record CreateSalesOrderLineModel(
    int? PartId,
    string Description,
    int Quantity,
    decimal UnitPrice,
    string? Notes);

File: qb-engineer-ui/src/app/features/sales-orders/models/create-sales-order-request.model.ts
import { CreateSalesOrderLineRequest } from './create-sales-order-line-request.model';

export interface CreateSalesOrderRequest {
  customerId: number;
  quoteId?: number;
  shippingAddressId?: number;
  billingAddressId?: number;
  creditTerms?: string;
  requestedDeliveryDate?: string;
  customerPO?: string;
  notes?: string;
  taxRate: number;
  lines: CreateSalesOrderLineRequest[];
}
